msg_queue.go

```diff
@@ -130,21 +228,30 @@ func stateToMsg(st State) MsgType {
 	}
 }
 
-type msgQueueImpl []*MessageReq
+type msgQueueImpl struct {
+	queue    []*MessageReq
+	updateCh chan struct{}
```

Comment:
Is this the same channel we have channel which we have in the `Pbft`? If it is, then it may be some issues, because we are reading it in the `getNextMessage` and here in `msg_queue` in `init` function.